Firmware/Marlin.h

```diff
@@ -316,6 +316,19 @@ bool printJobOngoing();
 
 bool printer_active();
 
+enum class PrinterStatus : uint8_t
+{
+    NotReady = 0,
+    IsReady = 1,
+    Idle = 2,
+    SDPrintingFinished = 3,
+    HostPrintingFinished = 4,
+    IsSDPrinting = 5,
+    IsHostPrinting = 6,
+};
+
+extern PrinterStatus printer_status;
```

Comment:
please do not add new extern variables :pray:

Reply:
Adding here the discussion we had two weeks ago:

Do not add new `extern` variables.

Instead create "get" and "set" functions. Preferably in a new file like `printer_status.cpp/h` or somewhere else than `Marlin.h`.

Then define `PrinterStatus printer_status` in the cpp file and use `static` keyword. I have something like this in mind:

```cpp
// cpp file
static PrinterStatus printer_status;

PrinterStatus GetPrinterStatus() {
    return printer_status;
}

PrinterStatus SetPrinterStatus(PrinterStatus status) {
    printer_status = status;
}
```

```cpp
// header file
enum class PrinterStatus : uint8_t
{
    NotReady = 0,
    IsReady = 1,
    Idle = 2,
    SDPrintingFinished = 3,
    HostPrintingFinished = 4,
    IsSDPrinting = 5,
    IsHostPrinting = 6,
};

PrinterStatus GetPrinterStatus();
PrinterStatus SetPrinterStatus(PrinterStatus status);
```